src/agent/onefuzz-agent/src/coordinator.rs

```diff
@@ -153,25 +156,25 @@ pub struct TaskInfo {
 pub trait ICoordinator: Downcast {
     async fn poll_commands(&mut self) -> Result<Option<NodeCommand>, PollCommandError>;
 
-    async fn emit_event(&mut self, event: NodeEvent) -> Result<()>;
+    async fn emit_event(&self, event: NodeEvent) -> Result<()>;
 
-    async fn can_schedule(&mut self, work: &WorkSet) -> Result<CanSchedule>;
+    async fn can_schedule(&self, work: &WorkSet) -> Result<CanSchedule>;
 }
 
 impl_downcast!(ICoordinator);
 
 #[async_trait]
 impl ICoordinator for Coordinator {
     async fn poll_commands(&mut self) -> Result<Option<NodeCommand>, PollCommandError> {
-        self.poll_commands().await
+        Coordinator::poll_commands(self).await
```

Comment:
This might be related to #2702 the existing code seemed to create an infinite loop in some case